src/main/java/org/broadinstitute/hellbender/utils/nio/SeekableByteChannelPrefetcher.java

```diff
@@ -0,0 +1,390 @@
+package org.broadinstitute.hellbender.utils.nio;
+
+import com.google.common.base.Stopwatch;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.NonWritableChannelException;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * SeekableByteChannelPrefetcher wraps an existing SeekableByteChannel to add prefetching.
+ * The prefetching is done on a different thread, so you can write simple code that repeatedly
+ * calls read() to get data, processes it, and then calls read again -- and yet this
+ * simple code overlaps computation and communication for you.
+ * (Of course this is only worthwhile if the underlying SeekableByteChannel doesn't already
+ * implement prefetching).
+ */
+public final class SeekableByteChannelPrefetcher implements SeekableByteChannel {
+
+    final SeekableByteChannel chan;
```

Comment:
Could you minimize the visibility of the class members.  I don't think we want anyone reaching in and touching our channel if we can avoid it.